lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -66,7 +66,7 @@ def purchase(amount, payment, options = {})
             add_payment_source(xml, payment)
             add_invoice(xml, options)
             add_customer_data(xml, payment, options)
-            add_retail_data(xml, payment)
+            add_market_type(xml, payment)
```

Comment:
Why aren't we passing market_type with auth's as well? looks like it's supported.